src/DataProtection/DataProtection/src/KeyManagement/XmlKeyManager.cs

```diff
@@ -423,7 +423,7 @@ IKey IInternalXmlKeyManager.CreateNewKey(Guid keyId, DateTimeOffset creationDate
         var possiblyEncryptedKeyElement = KeyEncryptor?.EncryptIfNecessary(keyElement) ?? keyElement;
 
         // Persist it to the underlying repository and trigger the cancellation token.
-        var friendlyName = string.Format(CultureInfo.InvariantCulture, "key-{0:D}", keyId);
+        var friendlyName = FormattableString.Invariant($"key-{keyId:D}");
```

Comment:
Can these use string.Create instead? (https://learn.microsoft.com/en-us/dotnet/csharp/tutorials/string-interpolation#how-to-create-a-result-string-using-the-invariant-culture)

Reply:
Is there an advantage?

Reply:
`string.Create` uses the interpolated string handler, so avoids [the intermediate allocation](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGADAAQY4B0A4gDYTACGtAlgF6MAuLEAdgNyFCGAMzkspAMKkA3oVILyYimQCyAChWkA1jACeASQAmASlIBeAHykAYtAC2XTo2C0YAZU5QWvAOaUDXgA3Rm9GXk51ABIAIl09OBl44xAAEQBfGJNBAkUlchwyADlNQp19YzMrAuJKSVguGHVJAFdaThbYQIAzCADg0JZwzlb2zpgAGlJY+MTkozTM7MJ0oA=).

To squeeze out more perf, in `string.Create` you could pass a stack-buffer to avoid the rent/return from array pool (if that stack-buffer is large enough though).

Reply:
This code is compiled for net462 which doesn't have string.Create.